File: py/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        rem = 0
        for ind, n in enumerate(nums):
            rem = target - n
            if rem in nums and nums.index(rem) != ind:
                return [ind, nums.index(rem)]

File: java/1_Two_Sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> map = new HashMap<Integer, Integer>();
        int complement;
        for (int i = 0; i<nums.length; i++) {
            complement = target - nums[i];  
            if (map.containsKey(complement)) {
                return new int[] {i, map.get(complement)};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No solution");
    }
}